### PR TITLE
Upgrade to SF 5.3

### PR DESCRIPTION
**Pull request by HedicGuibert**

### CODE REVIEW

**Comment by welcoMattic** [comment]
Did this just work??! There are not lots of deprecation messages that we should fix?

**Comment by HedicGuibert** [comment]
I might have missed them then ?

File: .php-cs-fixer.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...5928.php":3804833979,"migrations\/Version20210506150419.php":3971338163,"migrations\/Version20201216140149.php":1855282789,"migrations\/Version20190717142227.php":3391649362,"migrations\/Version20191211123106.php":2210492277,"migrations\/Version20191119074311.php":1754154068,"migrations\/Version20191205102223.php":1829892258,"migrations\/Version20210223150819.php":3987983751,"migrations\/Version20190718122216.php":1076743378,"migrations\/Version20210112170750.php":383791732,"migrations\/Version20190717090955.php":1414630119,"migrations\/Version20201216210507.php":864455377,"migrations\/Version20191218204843.php":4206486730,"migrations\/Version20191205180813.php":789068869,"migrations\/Version20210111100854.php":1753561441,"src\/DataTransformer\/ConferenceNameTransformer.php":1973394404,"src\/Form\/ConferenceDatalistType.php":4110323370,"migrations\/Version20210604120827.php":1818022867,"tests\/Controller\/User\/UserSubmitControllerTest.php":3449536852,"tests\/Controller\/User\/UserControllerTest.php":2550478441,"src\/Repository\/UserRepository.php":2657028335,"src\/Repository\/TalkRepository.php":3439604212,"tests\/Controller\/User\/UserParticipationControllerTest.php":2916586857}}
```
Comment (lyrixx) [inline .php-cs-fixer.cache:1]: Should not be commited!

File: config/packages/security.yaml [hunk 2]
```diff
@@ -12,21 +12,19 @@ security:
...
-                authenticators:
+            custom_authenticators:
                     - App\Security\Authenticator\LoginFormAuthenticator
                     - App\Security\Authenticator\GoogleAuthenticator
                     - App\Security\Authenticator\GitHubAuthenticator
             logout:
                 path: logout
                 target: conferences_list
-
+    enable_authenticator_manager: true
```
Comment (lyrixx) [inline config/packages/security.yaml:25]: Nice 👍🏼

File: src/Entity/User.php [hunk 3]
```diff
@@ -265,6 +266,11 @@ public function getUsername(): string
         return (string) $this->email;
```
Comment (lyrixx) [inline src/Entity/User.php:266]: Maybe it could be removed?

File: src/Security/Authenticator/LoginFormAuthenticator.php [hunk 4]
```diff
@@ -11,63 +11,81 @@
...
 
-    public function getCredentials(Request $request)
+    public function supports(Request $request): ?bool
     {
-        return [
-            'username' => $request->request->get('_username'),
-            'password' => $request->request->get('_password'),
-        ];
+        return 'login' === $request->attributes->get('_route')
+        && $request->isMethod('POST');
```
Comment (lyrixx) [inline src/Security/Authenticator/LoginFormAuthenticator.php:52]: CS is ugly :)

File: src/Security/Authenticator/LoginFormAuthenticator.php [hunk 5]
```diff
@@ -11,63 +11,81 @@
...
-        return $userProvider->loadUserByUsername($credentials['username']);
-    }
+        $username = $request->request->get('_username');
+        $password = $request->request->get('_password');
 
-    public function checkCredentials($credentials, UserInterface $user)
-    {
-        return $this->passwordManager->isPasswordValid($user->getPassword(), $credentials['password'] ?? '');
+        if (null === $username) {
+            throw new CustomUserMessageAuthenticationException('No username provided');
```
Comment (lyrixx) [inline src/Security/Authenticator/LoginFormAuthenticator.php:61]: Missing trailing dot

**Review by lyrixx — COMMENTED** [review]
Nice, thanks.

I left some comments

File: src/Entity/User.php (hunk 3 above)
Comment (HedicGuibert) [reply src/Entity/User.php:266]: I tried but the UseInterface still requires this method, same with `getSalt()`.
We ill be able to remove it in 5.4 if I remember correctly, or 6.0 maybe

File: phpunit.xml.dist.bak [hunk 6]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lyrixx) [inline phpunit.xml.dist.bak:1]: file to remove

**Review by lyrixx — APPROVED** [review]
Except one little comment, it's perfect.
You can merge as soon a fixed + green

**Pull request merged**